src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -589,6 +591,58 @@ class spell_warr_impending_victory : public SpellScript
     }
 };
 
+// 12950 - Improved Whirlwind (attached to 190411 - Whirlwind)
+class spell_improved_whirlwind : public SpellScript
+{
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo({ SPELL_WARRIOR_IMPROVED_WHIRLWIND, SPELL_WARRIOR_WHIRLWIND_CLEAVE_AURA })
+            && ValidateSpellEffect({ { spellInfo->Id, EFFECT_2 }, { SPELL_WARRIOR_WHIRLWIND_ENERGIZE, EFFECT_0 } });
```

Comment:
I would check if SPELL_WARRIOR_WHIRLWIND_CLEAVE_AURA is aura of type SPELL_AURA_ADD_FLAT_MODIFIER and if the misc is 17 (ChainTargets)

Reply:
I think I'd rather thave this be the same as in titanic rage script for now, and when a refactor happens to extract the code in own function, we could add a helper that also checks the aura effect and misc value